### DISCUSSION TITLE
paste function with overwrite

### DISCUSSION THREAD

**Discussion by og900aero** [gokcehan/lf#1748] [Ideas]
It would be nice if there was a another paste command (example paste-overwrite) that would not use dupfilefmt in case of duplication of the file, but would overwrite the file.

**Comment by TAforever** [comment]
```bash
cmd paste-overwrite %{{
    mode=$(head -1 ~/.local/share/lf/files)
    list=$(sed 1d ~/.local/share/lf/files)
    if [ $mode = 'copy' ]; then
        cp -r $list .
    elif [ $mode = 'move' ]; then
        mv $list .
    fi
    lf -remote 'send load'
    lf -remote 'send clear'
}}

**Reply by og900aero** [reply]
It's not good for me, because I use alternative copy methode, and your result doesnt show copy percentage. So would be nice if there is a base option to copy/move with overwrite.

**Comment by marendowski** [comment]
I use this to copy/cut files with progress:

```
# alternate paste workflow (select then move/copy)
cmd alt-paste &{{
    if [ -n "$fs" ]; then
        lf -remote "send $id $1"
    fi
    lf -remote "send $id paste"
}}

# show progress for file copying/cutting
# will overwrite files without any prompt!
cmd paste &{{
    set -- $(cat $HOME/.local/share/lf/files)
    mode="$1"
    shift
    if [ "$#" -lt 1 ]; then
        lf -remote "send $id echo no files selected"
        exit
    fi
    case "$mode" in
        copy)
            rsync --archive -hh --partial --info=stats1,progress2 --modify-window=1 "$@" . |
            stdbuf -i0 -o0 -e0 tr '\r' '\n' |
            while IFS= read -r line; do
                lf -remote "send $id echo $line"
            done
            ;;
        move)
            rsync --archive -hh --partial --info=stats1,progress2 --modify-window=1 --remove-source-files "$@" . |
            stdbuf -i0 -o0 -e0 tr '\r' '\n' |
            while IFS= read -r line; do
                lf -remote "send $id echo $line"
            done
            ;;
    esac
    lf -remote "send clear"
}}

map t alt-paste cut
map y alt-paste copy
```

You could use [pv](https://www.ivarch.com/programs/pv.shtml) to show better progress with `mv`/`cp`.

**Reply by og900aero** [reply]
> I use this to copy/cut files with progress:
> 
> ```
> # alternate paste workflow (select then move/copy)
> cmd alt-paste &{{
>     if [ -n "$fs" ]; then
>         lf -remote "send $id $1"
>     fi
>     lf -remote "send $id paste"
> }}
> 
> # show progress for file copying/cutting
> # will overwrite files without any prompt!
> cmd paste &{{
>     set -- $(cat $HOME/.local/share/lf/files)
>     mode="$1"
>     shift
>     if [ "$#" -lt 1 ]; then
>         lf -remote "send $id echo no files selected"
>         exit
>     fi
>     case "$mode" in
>         copy)
>             rsync --archive -hh --partial --info=stats1,progress2 --modify-window=1 "$@" . |
>             stdbuf -i0 -o0 -e0 tr '\r' '\n' |
>             while IFS= read -r line; do
>                 lf -remote "send $id echo $line"
>             done
>             ;;
>         move)
>             rsync --archive -hh --partial --info=stats1,progress2 --modify-window=1 --remove-source-files "$@" . |
>             stdbuf -i0 -o0 -e0 tr '\r' '\n' |
>             while IFS= read -r line; do
>                 lf -remote "send $id echo $line"
>             done
>             ;;
>     esac
>     lf -remote "send clear"
> }}
> 
> map t alt-paste cut
> map y alt-paste copy
> ```
> 
> You could use [pv](https://www.ivarch.com/programs/pv.shtml) to show better progress with `mv`/`cp`.

I want to use internal method.
